packages/requisitions/src/RequestRequisition/api/api.ts

```diff
@@ -123,6 +125,16 @@ export const getRequestQueries = (sdk: Sdk, storeId: string) => ({
       throw new Error('Record not found');
     },
   },
+  deleteLines: async (requestLines: RequestLineFragment[]) => {
+    const ids = requestLines.map(requestParser.toDeleteLine);
+    const result = await sdk.deleteRequestLines({ ids, storeId });
+
+    if (result.batchRequestRequisition.deleteRequestRequisitionLines) {
+      return result.batchRequestRequisition.deleteRequestRequisitionLines;
+    }
```

Comment:
too much? was wondering about adding in error handling into the hook and realised that the batch return needed some checking before it would be helpful

```suggestion
    if (result.batchRequestRequisition.deleteRequestRequisitionLines) {
      const failedLines =
        result.batchRequestRequisition.deleteRequestRequisitionLines.filter(
          line =>
            line.response.__typename === 'DeleteRequestRequisitionLineError'
        );
      if (failedLines.length === 0) {
        return result.batchRequestRequisition.deleteRequestRequisitionLines;
      }
    }
```

Reply:
Nice.

Yeah.. so if we went with throwing errors and utilising `onError` to catch and action, then we would likely want to throw whatever the errors were in the response array..

So maybe we could create a custom error object to wrap either a single response error or array of errors and catch that in `onError` 🤔